pkg/catalog/outbound_traffic_policies.go

```diff
@@ -66,12 +66,16 @@ func (mc *MeshCatalog) listOutboundTrafficPoliciesForTrafficSplits(sourceNamespa
 			Namespace: split.Namespace,
 		}
 
-		hostnames, err := mc.getServiceHostnames(svc, svc.Namespace == sourceNamespace)
+		locality := service.LocalCluster
+		if svc.Namespace == sourceNamespace {
+			locality = service.LocalNS
+		}
+		hostnames, err := mc.GetServiceHostnames(svc, locality)
```

Comment:
Same here, I don't think this logic is right. Refer to the comment below

Reply:
same comment as above